docs/spec/staking/state.md

```diff
@@ -167,8 +178,18 @@ the original redelegation has been completed.
 
 ```golang
 type Redelegation struct {
-    SourceShares           sdk.Dec     // amount of source shares redelegating
-    DestinationShares      sdk.Dec     // amount of destination shares created at redelegation
-    CompleteTime           int64       // unix time to complete redelegation
+	DelegatorAddr    sdk.AccAddress      // delegator
+	ValidatorSrcAddr sdk.ValAddress      // validator redelegation source operator addr
+	ValidatorDstAddr sdk.ValAddress      // validator redelegation destination operator addr
+	Entries          []RedelegationEntry // redelegation entries
+}
+
+type RedelegationEntry struct {
+	CreationHeight int64     // height which the redelegation took place
+	CompletionTime time.Time // unix time for redelegation completion
+	InitialBalance sdk.Coin  // initial balance when redelegation started
+	Balance        sdk.Coin  // current balance
+	SharesSrc      sdk.Dec   // amount of source shares redelegating
```

Comment:
Are these two fields used to define the functionality of "hopping" redelegations? I'm a bit confused on their semantics and what they mean.

Reply:
balance is the balance of the redelegation can be reduced by slashing (https://github.com/cosmos/cosmos-sdk/blob/develop/x/staking/keeper/slash.go#L194) 

Updated comments to     
    SharesSrc      sdk.Dec   // amount of source-validator shares removed by redelegation
    SharesDst      sdk.Dec   // amount of destination-validator shares created by redelegation

These are used for slashing - hopefully this clears up your understanding